src/ipc/ipc4/handler.c

```diff
@@ -465,7 +465,7 @@ static int ipc4_process_chain_dma(struct ipc4_message_request *ipc4)
 	bool delay = false;
 	int ret;
 
-	memcpy_s(&cdma, sizeof(cdma), ipc4, sizeof(cdma));
+	memcpy_s(&cdma, sizeof(cdma), ipc4, sizeof(*ipc4));
```

Comment:
I think this is just obfuscation of what we do...
```c
cdma.primary.dat = ipc4.primary.dat;
cdma.extension.dat = ipc4.extension.dat;
```
is done here or the ipc4 point to be re-casted to chain dma struct:
```c
struct ipc4_chain_dma *cdma = (struct ipc4_chain_dma *) ipc4;
```
it could be safer to have a local copy, but I don't think it matter, using the same memory is fine here.